packages/api/src/datasources/projects.ts

```diff
@@ -19,6 +21,56 @@ const getSortByAggregation = (direction = 'DESC') => ({
   },
 });
 
+const addHookIdsToProjectHooks = (project) => {
+  if (project && project.hooks) {
+    project.hooks = project.hooks.map((hook) => {
+      hook.hookId = hook.hookId || uuid();
+      return hook;
+    });
+  }
+  return project;
+};
+
+const removeUnusedHookDataFromProject = (project) => {
+  if (project && project.hooks) {
+    project.hooks = project.hooks.map((hook) => {
+      if (hook.hookType === hookTypes.GENERIC_HOOK) {
+        delete hook.githubToken;
+      }
+      if (hook.hookType === hookTypes.GITHUB_STATUS_HOOK) {
+        delete hook.headers;
+        delete hook.hookEvents;
+      }
+      return hook;
+    });
+  }
+  return project;
+};
+
+const restoreGithubTokensOnGithubHooks = async (
+  updatedProject,
+  getProjectById
+) => {
+  const oldProject = await getProjectById(updatedProject.projectId);
+
+  // This is to ensure that we keep github tokens when the user only updaing the url
+  if (updatedProject && updatedProject.hooks) {
+    updatedProject.hooks = updatedProject.hooks.map((hook) => {
+      if (!hook.githubToken) {
+        const oldhook =
+          oldProject &&
+          oldProject.hooks &&
+          oldProject.hooks.find((oldHook) => oldHook.hookId === hook.hookId);
```

Comment:
```suggestion
      if (!hook.githubToken) {
        const oldhook = oldProject.?hooks?.find((oldHook) => oldHook.hookId === hook.hookId);
```

Reply:
Totally. I would have but ... optional chaining is only setup on dashboard. I attempted to implement it on director and typescript pooped the bed. So kinda backed off it for now. If you wanna get it functional I would be super happy to updated all these cause optional chaining is way more readable.

Reply:
I am pretty sure it is working now

Reply:
It is! Huzzah! I just fixed these all locally.